main.js

```diff
@@ -442,6 +443,9 @@ let drawObstacleBoundingBox = function(obstacle) {
 };
 
 let is_overlapping = function(object1, object2) {
+  if (!object2.obst) {
```

Comment:
As you already found out, the default of a property is `undefined` which is treated as a _falsy_ value and acts like `false`. In other words (again as you already concluded, just writing this down here for reference) you need to reverse the logic here if you don't want to add this new property to all obstacles.